src/components/FieldOptionEditor.tsx

```diff
@@ -0,0 +1,43 @@
+import * as React from 'react';
+import * as data from '../data';
+
+interface IProps {
+    field: data.IField;
+    registry: data.FieldRegistry;
+    onChange: (field: data.IField) => void;
```

Comment:
Discussed this offline. We may need to think this a bit more.

The questions is whether the FieldOptionEditor should manage the state of all fields.